modules/forum/src/main/Post.scala

```diff
@@ -28,6 +35,26 @@ case class Post(
   def isTeam = categId startsWith teamSlug("")
 
   def isStaff = categId == "staff"
+
+  def canStillBeEdited() = {
+    val firstPost = editHistory.lastOption map (_.createdAt) getOrElse createdAt
+    firstPost.plus(permitEditsFor.toMillis).isAfter(DateTime.now)
+  }
+
+  def editedTooSoonAfterLastEdit() = {
+    editHistory.length > 3 && createdAt.plus(coolOffBetweenEdits.toMillis).isAfter(DateTime.now)
+  }
+
+  def canBeEditedBy(editingId: Option[String]) : Boolean = editingId.isDefined && editingId == userId
+
+  def editPost(updated: DateTime, newText: String) : Post = {
+    val oldVersion = new OldVersion(text, createdAt)
+    val history = oldVersion :: editHistory
+
+    this.copy(editHistory = history, text = newText, createdAt = updated)
```

Comment:
By convention in lila, `createdAt` is immutable. Objects that are updated get an `updatedAt` field.